### PR TITLE
Add RC smoke tester that checks basic functionality

### PR DESCRIPTION
**Pull request by s1monw**
This RC smoke tester runs:
1. Downloads the tar.gz & zip file from the staging URL
2. Verifies it's sha1 hashes and GPG signatures against the release key
3. Installs all official plugins
4. Starts one node an checks:
   -- if it runs with Java 1.7
   -- if the build hash given is the one that is returned by the status response
   -- if the build is a release version and not a snapshot version
   -- if all plugins are loaded
   -- if the status reponse returns the correct version

### CODE REVIEW

**Comment by s1monw** [comment]
@spinscale I basically put all the things that I do anyway into a script - we can improve it as we go I guess

File: dev-tools/smoke_test_rc.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+# an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+# Smoke-tests a release candidate
+#
+# 1. Downloads the tar.gz & zip file from the staging URL
+# 2. Verifies it's sha1 hashes and GPG signatures against the release key
+# 3. Installs all official plugins
+# 4. Starts one node an checks:
```
Comment (rjernst) [inline dev-tools/smoke_test_rc.py]: nit: an -> and

File: dev-tools/smoke_test_rc.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+import urllib.request
+import hashlib
+import time
+import socket
+import json
+
+from prepare_release_candidate import run
+from http.client import HTTPConnection
+
+DEFAULT_PLUGINS = ["analysis-icu",
```
Comment (rjernst) [inline dev-tools/smoke_test_rc.py]: Nothing to do here for this PR, but how can we keep this up to date? We already have this list in like 3 other places it seems...

File: dev-tools/smoke_test_rc.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+try:
+  JAVA_HOME = os.environ['JAVA_HOME']
+except KeyError:
+  raise RuntimeError("""
+  Please set JAVA_HOME in the env before running release tool
+  On OSX use: export JAVA_HOME=`/usr/libexec/java_home -v '1.7*'`""")
+
+try:
+  JAVA_HOME = os.environ['JAVA7_HOME']
```
Comment (rjernst) [inline dev-tools/smoke_test_rc.py]: Why have this check? I know I keep this variable around, as well as one for java 8, and set JAVA_HOME based on it when I want to switch.

File: dev-tools/smoke_test_rc.py [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+                      help='The Elasticsearch Version to smoke-tests', required=True)
+  parser.add_argument('--hash', '-s', dest='hash', default=None, required=True,
+                      help='The sha1 short hash of the git commit to smoketest')
+  parser.set_defaults(hash=None)
+  parser.set_defaults(version=None)
+  args = parser.parse_args()
+  version = args.version
+  hash = args.hash
+  files = [
+    'org/elasticsearch/distribution/tar/elasticsearch/2.0.0-beta1/elasticsearch-2.0.0-beta1.tar.gz',
```
Comment (rjernst) [inline dev-tools/smoke_test_rc.py]: Is this script meant only for the beta? Is the intention to generify it in the future?

File: dev-tools/smoke_test_rc.py (hunk 2 above)
Comment (rjernst) [inline dev-tools/smoke_test_rc.py]: Maybe we could programmatically build this from listing the plugins dir?

File: dev-tools/smoke_test_rc.py [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+    es_run_path = os.path.join(tmp_dir, 'elasticsearch-%s' % (release), 'bin/elasticsearch')
+    print('  Smoke testing package [%s]' % release_file)
+    es_plugin_path = os.path.join(tmp_dir, 'elasticsearch-%s' % (release),'bin/plugin')
+    plugin_names = {}
+    for plugin  in plugins:
+      print('  Install plugin [%s]' % (plugin))
+      run('%s; %s -Des.plugins.staging=true %s %s' % (java_exe(), es_plugin_path, 'install', plugin))
+      plugin_names[plugin] = True
+    print('  Starting elasticsearch deamon from [%s]' % os.path.join(tmp_dir, 'elasticsearch-%s' % release))
+    run('%s; %s -Des.node.name=smoke_tester -Des.cluster.name=prepare_release -Des.script.inline=on -Des.script.indexed=on -Des.repositories.url.allowed_urls=http://snapshot.test* %s -Des.pidfile=%s'
```
Comment (rjernst) [inline dev-tools/smoke_test_rc.py]: why enabling dynamic scripting? I don't see it used?

**Comment by rjernst** [comment]
I left a couple comments, I like this a lot!!

**Comment by rjernst** [comment]
BTW, none of the comments I left are critical. This LGTM regardless of the responses.

File: dev-tools/smoke_test_rc.py (hunk 3 above)
Comment (s1monw) [inline dev-tools/smoke_test_rc.py]: yeah I do this too :) so I added it but hey I think we can just remove it!

File: dev-tools/smoke_test_rc.py (hunk 2 above)
Comment (s1monw) [inline dev-tools/smoke_test_rc.py]: I will open an issue for this

Comment (s1monw) [inline dev-tools/smoke_test_rc.py]: I added https://github.com/elastic/elasticsearch/issues/13096

**Pull request merged**